fda-model/alternativeComparisonReport/AlternativeComparisonReport.cs

```diff
@@ -13,37 +13,47 @@ namespace HEC.FDA.Model.alternativeComparisonReport
         public event ProgressReportedEventHandler ProgressReport;
+        #endregion
+
+        #region Constructor 
+        public AlternativeComparisonReport()
+        {
+            MessageHub.Register(this);
+        }
+        #endregion 
 
         public AlternativeComparisonReportResults ComputeAlternativeComparisonReport(IProvideRandomNumbers randomProvider, ConvergenceCriteria convergenceCriteria, AlternativeResults withoutProjectAlternativeResults, List<AlternativeResults> withProjectAlternativesResults)
         {
             ReportMessage(this, new MessageEventArgs(new ErrorMessage("Starting alternative comparison report compute", ErrorLevel.Info)));
             ReportProgress(this, new ProgressReportEventArgs(10));
 
             MessageEventArgs beginComputeMessageArgs = new MessageEventArgs(new Message("The alternative results are being processed for the alternative comparison report."));
-            withoutProjectAlternativeResults.ReportMessage(withoutProjectAlternativeResults, beginComputeMessageArgs);
+            ReportMessage(this, beginComputeMessageArgs);
             List<ConsequenceDistributionResults> aaeqResults = ComputeDistributionOfAAEQDamageReduced(randomProvider, convergenceCriteria, withoutProjectAlternativeResults, withProjectAlternativesResults);
             MessageEventArgs aaeqResultsMessageArgs = new MessageEventArgs(new Message("The distributions of AAEQ Damage Reduced for the given with-project conditions have been computed."));
-            withoutProjectAlternativeResults.ReportMessage(withoutProjectAlternativeResults, aaeqResultsMessageArgs);
+            ReportMessage(this, aaeqResultsMessageArgs);
             List<ConsequenceDistributionResults> baseYearEADResults = ComputeDistributionEADReducedBaseYear(randomProvider, convergenceCriteria, withoutProjectAlternativeResults, withProjectAlternativesResults);
             MessageEventArgs baseYearEADReducedMessageArgs = new MessageEventArgs(new Message("THe distributions of base year EAD reduced for the given with-project conditions have been computed."));
-            withoutProjectAlternativeResults.ReportMessage(withoutProjectAlternativeResults, baseYearEADReducedMessageArgs);
+            ReportMessage(this, baseYearEADReducedMessageArgs);
             List<ConsequenceDistributionResults> futureYearEADResults = ComputeDistributionEADReducedFutureYear(randomProvider, convergenceCriteria, withoutProjectAlternativeResults, withProjectAlternativesResults);
             MessageEventArgs futureYearEADReducedMessageArgs = new MessageEventArgs(new Message("The distributions of future year EAD reduced for the given with-project conditions have been computed."));
-            withoutProjectAlternativeResults.ReportMessage(withoutProjectAlternativeResults, futureYearEADReducedMessageArgs);
+            ReportMessage(this, futureYearEADReducedMessageArgs);
 
+            //TODO: Fixed the hacked in progress reporting 
```

Comment:
Do we need this todo statement? It is past tense. Is there anything that needs to be done?